x-pack/plugins/apm/public/components/shared/FilterBar/DatePicker.tsx

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import datemath from '@elastic/datemath';
+import {
+  CommonProps,
+  EuiSuperDatePicker,
+  EuiSuperDatePickerProps
+} from '@elastic/eui';
+import { memoize } from 'lodash';
+import React from 'react';
+import { connect } from 'react-redux';
+import { RouteComponentProps, withRouter } from 'react-router-dom';
+import { updateTimePicker } from 'x-pack/plugins/apm/public/store/urlParams';
+import { fromQuery, toQuery } from '../Links/url_helpers';
+
+interface Props extends RouteComponentProps {
+  dispatchUpdateTimePicker: typeof updateTimePicker;
+}
+
+interface DatePickerParams {
+  rangeFrom: string;
+  rangeTo: string;
+  refreshPaused: boolean;
+  refreshInterval: number;
+}
+
+// TODO this can be removed when EUI PR lands: https://github.com/elastic/eui/pull/1574
+declare module '@elastic/eui' {
```

Comment:
Until then, should we have it in the "typings" folder?

Reply:
Eg. like this: https://github.com/elastic/kibana/blob/00fbc2572efe519dd2169b3e21845b1397ecdf21/x-pack/plugins/apm/typings/numeral.d.ts